src/main/java/org/apache/bcel/generic/ObjectType.java

```diff
@@ -34,38 +34,38 @@
 @AnnotatedFor({"signature"})
 public class ObjectType extends ReferenceType {
 
-    private final @BinaryNameForNonArray String class_name; // Class name of type
+    private final @BinaryName String class_name; // Class name of type
 
     /**
      * @since 6.0
      */
-    // TODO: actualy type is @BinaryNameForNonArray =
+    // TODO: actualy type is @BinaryName =
     // @ClassGetName-for-nonarray, but that type requires warning
     // suppressions at call sites, so just use @ClassGetName.
-    public static ObjectType getInstance(final /*@ BinaryNameForNonArray*/ @ClassGetName String class_name) {
+    public static ObjectType getInstance(final /*@ BinaryName*/ @ClassGetName String class_name) {
         return new ObjectType(class_name);
     }
 
     /**
      * @param class_name fully qualified class name, e.g. java.lang.String
      */
-    // TODO: buggy documentation, is a @BinaryNameForNonArray (differs from @FullyQualifiedName for inner classes).
-    // TODO: actualy type is @BinaryNameForNonArray =
+    // TODO: buggy documentation, is a @BinaryName (differs from @FullyQualifiedName for inner classes).
+    // TODO: actualy type is @BinaryName =
```

Comment:
You should write annotations that express the code's requirements/contracts.  If the method might crash when given a `@ClassGetName`, then don't annotate it as `@ClassGetName`.  Also please clean up the comments, which are confusing (I'm not sure which documentation is buggy and what remains to be done).  Thanks!

(Friendly reminder:  please do not reply to GitHub conversations via email, because it leads to comments being placed in the wrong location.  Instead, reply to GitHub conversations from the GitHub website.)

Reply:
I've pushed a commit with the proposed changes.